packages/apollo-gateway/src/index.ts

```diff
@@ -40,6 +45,13 @@ interface GatewayConfigBase {
   // should cutover to use the new option when it's built.
   __exposeQueryPlanExperimental?: boolean;
   buildService?: (definition: ServiceEndpointDefinition) => GraphQLDataSource;
+
+  // experimental observability callbacks
+  experimental_didResolveQueryPlan?: DidResolveQueryPlanCallback;
+  experimental_didFailComposition?: DidFailCompositionCallback;
+  experimental_updateServiceDefinitions?: LoadServiceDefinitions;
```

Comment:
nit: hmm this says `updateServiceDefinition` and then it's typed to a `LoadServiceDefinitions` type which sounds like two different things: loading vs updating 

maybe they it should either be load or update not both?

Reply:
Also not sure if this is supporting back-compat or something we just need to clean up?

Reply:
It's a bit of both. `loadServiceDefinitions` is the name of the default service definition loading function (and I believe more accurate to what the function is doing), but the original proposal for this function was so it would be named `experimental_updateServiceDefinitions`. Do you think we should rename the experimental function to be `experimental_loadServiceDefinitions` @zionts @caydie-tran?